### PR TITLE
[Pb 8738] - Backup Schedule | Ansible Module

### PR DESCRIPTION
**Pull request by parveenp-px**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** (optional)
https://purestorage.atlassian.net/browse/PB-8738
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: ansible-collection/plugins/modules/px_backup/backup_schedule.py [hunk 1]
```diff
@@ -0,0 +1,494 @@
...
+
+def enumerate_backup_schedules(module, client):
+    """List all backup locations"""
+    params={}
+    if module.params.get('backup_location_ref'):
+        backup_location_ref = module.params['backup_location_ref']
+        params['backup_location_ref.name'] = backup_location_ref.get('name')
+        params['backup_location_ref.uid'] = backup_location_ref.get('uid')
+
+    if module.params.get('enumerate_options'):
```
Comment (t-nidhin) [inline ansible-collection/plugins/modules/px_backup/backup_schedule.py:225]: `enumerate_options`  how is this imported without module_args?

Comment (t-nidhin) [reply ansible-collection/plugins/modules/px_backup/backup_schedule.py:225]: and it's child params `labels`, `max_objects`,`time_range`, etc

File: ansible-collection/plugins/modules/px_backup/backup_schedule.py [hunk 2]
```diff
@@ -0,0 +1,494 @@
...
+        uid=dict(type='str', required=False),
+        owner=dict(type='str', required=False),
+        schedule_policy=dict(type='str', required=False),
+        reclaim_policy=dict(type='str', required=False),
+        backup_location=dict(type='str', required=False),
+        cluster=dict(type='str', required=False),
+        pre_exec_rule=dict(type='str', required=False),
+        post_exec_rule=dict(type='str', required=False),
+        csi_snapshot_class_name=dict(type='str', required=False),
+        backup_type=dict(type='str', required=False),
```
Comment (t-nidhin) [inline ansible-collection/plugins/modules/px_backup/backup_schedule.py:339]: add choices: `Invalid`, `Generic`, and `Normal`

File: ansible-collection/plugins/modules/px_backup/backup_schedule.py [hunk 3]
```diff
@@ -0,0 +1,494 @@
...
+        direct_kdmp=dict(type='bool', required=False),
+        include_resources=dict(type='list', elements='dict', options=dict(
+                name=dict(type='str'),
+                namespace=dict(type='str'),
+                group=dict(type='str'),
+                kind=dict(type='str'),
+                version=dict(type='str'),
+
+            )),
+        backup_object_type=dict(
```
Comment (t-nidhin) [inline ansible-collection/plugins/modules/px_backup/backup_schedule.py:352]: add choices:
```
    message BackupObjectType {
        Type type = 1;
         enum Type {
             Invalid = 0;
             All = 1; // for All application backup
             VirtualMachine = 2; // virtualMachine for VirtualMachine specific backup
         }
     }
```

File: ansible-collection/plugins/modules/px_backup/backup_schedule.py [hunk 4]
```diff
@@ -0,0 +1,494 @@
...
+    if module.params.get('label_selectors'):
+        backup_schedule_request['label_selectors'] = module.params['label_selectors']
+        
+    if module.params.get('volume_snapshot_class_mapping'):
+        backup_schedule_request['volume_snapshot_class_mapping'] = module.params['volume_snapshot_class_mapping']
+
+    return backup_schedule_request
+
+def run_module():
+    module_args = dict(
```
Comment (t-nidhin) [inline ansible-collection/plugins/modules/px_backup/backup_schedule.py:324]: `exclude_resource_types` <-- very important

File: ansible-collection/plugins/modules/px_backup/backup_schedule.py [hunk 5]
```diff
@@ -0,0 +1,494 @@
...
+        api_url=dict(type='str', required=True),
+        token=dict(type='str', required=True, no_log=True),
+        operation=dict(type='str', choices=['CREATE', 'UPDATE', 'DELETE','INSPECT_ALL','INSPECT_ONE'], default='CREATE'),
```
Comment (t-nidhin) [inline ansible-collection/plugins/modules/px_backup/backup_schedule.py:327]: mark it as required, remove default

File: ansible-collection/plugins/modules/px_backup/backup_schedule.py [hunk 6]
```diff
@@ -0,0 +1,494 @@
...
+        name=dict(type='str', required=False),
+        org_id=dict(type='str', required=True),
+        uid=dict(type='str', required=False),
+        owner=dict(type='str', required=False),
+        schedule_policy=dict(type='str', required=False),
+        reclaim_policy=dict(type='str', required=False),
```
Comment (t-nidhin) [inline ansible-collection/plugins/modules/px_backup/backup_schedule.py:333]: reclaim policy: choices=['Invalid', 'Delete', 'Retain']

File: ansible-collection/plugins/modules/px_backup/backup_schedule.py [hunk 7]
```diff
@@ -0,0 +1,494 @@
...
+        response = client.make_request(
+            'DELETE',
+            f"v1/backupschedule/{module.params['org_id']}/{module.params['name']}"
+        )
+        return response, True
+    except Exception as e:
+        module.fail_json(msg=f"Failed to delete Backup Schedule: {str(e)}")
+
+def backup_schedule_request_body(module):
+    """Build the Cloud Credential request object"""
```
Comment (t-nidhin) [inline ansible-collection/plugins/modules/px_backup/backup_schedule.py:275]: typo cloud cred request body..

File: ansible-collection/plugins/modules/px_backup/backup_schedule.py [hunk 8]
```diff
@@ -0,0 +1,494 @@
...
+    backup_schedule_request = {
+        "metadata": {
+            "name": module.params['name'],
+            "org_id": module.params['org_id'],
+            "owner": module.params['owner']
+        },
+        "schedule_policy": module.params['schedule_policy'],
```
Comment (t-nidhin) [inline ansible-collection/plugins/modules/px_backup/backup_schedule.py:282]: change to `"schedule_policy_ref": module.params.get('schedule_policy_ref'),`

Comment (t-nidhin) [reply ansible-collection/plugins/modules/px_backup/backup_schedule.py:282]: schedule_policy is deprecated

File: ansible-collection/plugins/modules/px_backup/backup_schedule.py [hunk 9]
```diff
@@ -0,0 +1,494 @@
...
+        "reclaim_policy": module.params['reclaim_policy'],
+        "backup_location": module.params['backup_location'],
```
Comment (t-nidhin) [inline ansible-collection/plugins/modules/px_backup/backup_schedule.py:284]: this is deprecated, use `"backup_location_ref": module.params.get('backup_location_ref'),`

File: ansible-collection/plugins/modules/px_backup/backup_schedule.py [hunk 10]
```diff
@@ -0,0 +1,494 @@
...
+        "cluster": module.params['cluster'],
```
Comment (t-nidhin) [inline ansible-collection/plugins/modules/px_backup/backup_schedule.py:285]: "cluster_ref": module.params.get('cluster_ref'),

**Review by t-nidhin — CHANGES_REQUESTED** [review]

File: ansible-collection/plugins/modules/px_backup/backup_schedule.py (hunk 8 above)
Comment (parveenp-px) [reply ansible-collection/plugins/modules/px_backup/backup_schedule.py:333]: removed

**Pull request merged**